flink/v1.17/flink/src/main/java/org/apache/iceberg/flink/source/split/FileSequenceNumberBasedComparator.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.flink.source.split;
+
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+
+public class FileSequenceNumberBasedComparator
+    implements SerializableComparator<IcebergSourceSplit> {
+  @Override
+  public int compare(IcebergSourceSplit o1, IcebergSourceSplit o2) {
+    Preconditions.checkArgument(
+        o1.task().files().size() == 1 && o2.task().files().size() == 1,
+        "Could not compare combined task. Please use 'split-open-file-cost' to prevent combining multiple files to a split");
+
+    Long seq1 = o1.task().files().iterator().next().file().fileSequenceNumber();
+    Long seq2 = o2.task().files().iterator().next().file().fileSequenceNumber();
+
+    Preconditions.checkNotNull(
+        seq1, "Invalid file sequence number for {}. Only V2 table format is supported", o1);
```

Comment:
convention is to call out what exactly is invalid (like null). here is what I have. 
```
Invalid file sequence number: null. Expected splits from V2 table format with valid file sequence number
```

Reply:
But printing out `null` here does not tell the user/administrator which split is causing the issue.
Whereas adding the split will help fixing the problem

Reply:
agree with what you side regarding which split causing the issue. but we will need to call out what exactly is invalid (`null` in this case). maybe print the split in the end of the formatted string
```
Invalid file sequence number: null. Doesn't support splits written with V1 format: %s
```